docs/encryption.md

```diff
@@ -5,31 +5,183 @@ is supported by the Operator since version 1.1.0.
+    ```bash
+    $ kubectl deploy -f deploy/cr.yaml
+    ```
+
+## <a name="using-vault"></a>Using HashiCorp Vault storage for encryption keys
+
+Starting from the version 1.13, the Operator supports using [HashiCorp Vault](https://www.vaultproject.io/) storage for encryption keys - a universal, secure and reliable way to store and distribute secrets without depending on the operating system, platform or cloud provider.
+
+!!! warning
+
+    Vault integration has technical preview status and is not yet recommended
+    for production environments.
+
+The Operator will use Vault if the `deploy/cr.yaml` configuration file contains
+the following items:
+
+* a `secrets.vault` key equal to the name of a specially created Secret,
+* `configuration` keys for mongod and config servers with a number of
+    Vault-specific options.
+
+The Operator itself neither installs Vault, nor configures it; both operations 
+should be done manually, as described in the following parts.
+
+### Installing Vault
+
+The following steps will deploy Vault on Kubernetes with the [Helm 3 package manager](https://helm.sh/). Other Vault installation methods should also work, so the instruction placed here is not obligatory and is for illustration purposes. Read more about installation in Vault’s [documentation](https://www.vaultproject.io/docs/platform/k8s).
+
+1. Add helm repo and install:
+
+    ```bash
+    $ helm repo add hashicorp https://helm.releases.hashicorp.com
+    "hashicorp" has been added to your repositories
+
+    $ helm install vault hashicorp/vault
+    ```
+
+2. After installation, Vault should be first initialized and then *unsealed*.
+    Initializing Vault is done with the following commands:
+
+    ```bash
+    $ kubectl exec -it pod/vault-0 -- vault operator init -key-shares=1 -key-threshold=1 -format=json > /tmp/vault-init
+    $ unsealKey=$(jq -r ".unseal_keys_b64[]" < /tmp/vault-init)
+    ```
+
+    To unseal Vault, execute the following command **for each Pod** of Vault
+    running:
+
+    ```bash
+    $ kubectl exec -it pod/vault-0 -- vault operator unseal "$unsealKey"
+    ```
+
+### Configuring Vault
+
+1. First, you should enable secrets within Vault. For this you will need a [Vault token](https://www.vaultproject.io/docs/concepts/tokens).
+    Percona Server for MongoDB can use any regular token which allows all operations
+    inside the secrets mount point. In the following example we are using the
+    *root token* to be sure the permissions requirement is met, but actually
+    there is no need in root permissions. We don’t recommend using the root token
+    on the production system.
+
+    ```bash
+    $ cat /tmp/vault-init | jq -r ".root_token"
+    ```
+
+    The output will show you the token:
+
+    ```text
+    s.VgQvaXl8xGFO1RUxAPbPbsfN
+    ```
+
+    Now login to Vault with this token and enable the key-value secret engine:
+
+    ```bash
+    $ kubectl exec -it vault-0 -- /bin/sh
+    $ vault login s.VgQvaXl8xGFO1RUxAPbPbsfN
```

Comment:
We may add the command output @spron-in what do you think? 
`marukovich@Natalias-MacBook-Pro percona-version-service [psmdb-1.13.0_update_operator_image] $ kubectl exec -it vault-0 -- /bin/sh
/ $ vault login hvs.7hTVYdJjROkpDJAxSLl3HKH9
Success! You are now authenticated. The token information displayed below
is already stored in the token helper. You do NOT need to run "vault login"
again. Future Vault requests will automatically use this token.

Key                  Value
---                  -----
token                hvs.7hTVYdJjROkpDJAxSLl3HKH9
token_accessor       iMGp477aReYkPBWrR42Z3L6R
token_duration       ∞
token_renewable      false
token_policies       ["root"]
identity_policies    []
policies             ["root"]`